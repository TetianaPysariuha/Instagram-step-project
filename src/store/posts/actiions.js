export const GET_POSTS = 'GET_POSTS';
export const GET_POST_BY_ID = 'GET_POST_BY_ID';
export const GET_POSTS_BY_USER_ID = 'GET_POSTS_BY_USER_ID';
export const ADD_POST = 'ADD_POST';
export const UPDATE_POST = 'UPDATE_POST';
export const DELETE_POST = 'DELETE_POST';
export const SHOW_MORE_CHANGE = 'SHOW_MORE_CHANGE';
export const GET_NEW_PAGE_POSTS = 'GET_NEW_PAGE_POSTS';
export const CLEAR_POSTS = 'CLEAR_POSTS';
export const GET_FAVORITE_POSTS = 'GET_FAVORITE_POSTS';
