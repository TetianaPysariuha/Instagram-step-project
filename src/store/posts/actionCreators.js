import {
  GET_POSTS, GET_POST_BY_ID, GET_POSTS_BY_USER_ID, ADD_POST, UPDATE_POST, DELETE_POST, SHOW_MORE_CHANGE, GET_NEW_PAGE_POSTS, CLEAR_POSTS, GET_FAVORITE_POSTS, CHANGE_ISOPENPOST,
} from './actiions';

const SERVER_URL = 'https://nameless-lake-66137.herokuapp.com';

export const getPosts = () => async (dispatch) => {
  const posts = await fetch(`${SERVER_URL}/posts`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POSTS, payload: posts });
};

export const clearPosts = () => ({ type: CLEAR_POSTS });

export const loadNewPagePosts = (payload) => async (dispatch) => {
  if (payload.start !== null && payload.end) {
    const response = await fetch(`${SERVER_URL}/posts/quantity/${payload.start}&${payload.end}`).then((res) => res.json()).then((data) => data);
    dispatch({ type: GET_NEW_PAGE_POSTS, payload: { data: response.data, countAll: response.totalCount } });
  } else {
    console.error(`For loafing posts should be start, end and userId parametres. Received: start=${payload.start},  end=${payload.end}`);
  }
};

export const getPostById = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POST_BY_ID, payload: post });
};

export const getPostsByUserId = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/user/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_POSTS_BY_USER_ID, payload: post });
};

export const addNewPost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts`, {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: ADD_POST, payload: post });
};

export const updatePost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload.id}`, {
    method: 'PUT',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload.data),
  }).then((res) => res.json()).then((data) => data);
  dispatch({ type: UPDATE_POST, payload: post });
};

export const deletePost = (payload) => async (dispatch) => {
  const post = await fetch(`${SERVER_URL}/posts/${payload}`, {
    method: 'DELETE',
    mode: 'cors',
    headers: { 'Content-Type': 'aplication/json' },
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: DELETE_POST, payload: post });
};

export const showMoreChange = (payload) => ({ type: SHOW_MORE_CHANGE, payload });

export const getFavoritePostsByUserId = (payload) => async (dispatch) => {
  const posts = await fetch(`${SERVER_URL}/posts/favorite/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_FAVORITE_POSTS, payload: posts });
};

export const changeIsOpenPost = (payload) => ({ type: CHANGE_ISOPENPOST, payload });
