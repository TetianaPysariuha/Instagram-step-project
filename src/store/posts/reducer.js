/* eslint-disable no-underscore-dangle */
import {
  GET_POSTS, GET_POST_BY_ID, GET_POSTS_BY_USER_ID, ADD_POST, UPDATE_POST, DELETE_POST, SHOW_MORE_CHANGE, GET_NEW_PAGE_POSTS, CLEAR_POSTS, GET_FAVORITE_POSTS, CLEAR_CURRENT_POST,
} from './actiions';

const initialState = {
  posts: [],
  currentPost: {},
  showMoreComments: [],
  page: 0,
  postsOnPage: 3,
  totalPostsOnServer: 0,
  favorites: [],
};

// the rule is switched off because state shoul be first in props
// eslint-disable-next-line default-param-last
const postsReduser = (state = initialState, action) => {
  switch (action.type) {
    case GET_POSTS: {
      return { ...state, posts: action.payload };
    }
    case CLEAR_POSTS: {
      return {
        ...state, posts: [], page: 0, totalPostsOnServer: 0,
      };
    }
    case CLEAR_CURRENT_POST: {
      return { ...state, currentPost: {} };
    }
    case GET_NEW_PAGE_POSTS: {
      if (action.payload.data.length > 0 && state.posts.length < action.payload.countAll) {
        const newPosts = [...state.posts, ...action.payload.data];
        const newPage = state.page + 1;
        return {
          ...state, posts: newPosts, page: newPage, totalPostsOnServer: action.payload.countAll,
        };
      }
      return state;
    }
    case GET_POST_BY_ID: {
      return { ...state, currentPost: action.payload };
    }
    case GET_POSTS_BY_USER_ID: {
      return { ...state, currentPost: action.payload };
    }
    case GET_FAVORITE_POSTS: {
      return { ...state, favorites: action.payload };
    }
    case ADD_POST: {
      /* const newPosts = [...state.posts];
      newPosts.push(action.payload);
      return { ...state, posts: newPosts }; */
      return state;
    }
    case UPDATE_POST: {
      if (action.payload.status === 'success') {
        const newPosts = [...state.posts];
        newPosts.splice(newPosts.findIndex((el) => el._id === action.payload.data._id), 1, action.payload.data);
        return { ...state, posts: newPosts, currentPost: action.payload.data };
      }
      return state;
    }
    case DELETE_POST: {
      if (action.payload.status === 'success') {
        const newPosts = state.posts.filter((post) => post._id !== action.payload.data._id);
        return { ...state, posts: newPosts };
      }
      return state;
    }
    case SHOW_MORE_CHANGE: {
      const showMoreComments = [...state.showMoreComments];
      if (showMoreComments.filter((el) => el === action.payload.postId).length > 0) {
        const newShowMoreComments = showMoreComments.filter((el) => el !== action.payload.postId);
        return { ...state, showMoreComments: newShowMoreComments };
      }
      showMoreComments.push(action.payload.postId);
      return { ...state, showMoreComments };
    }
    default: {
      return state;
    }
  }
};

export default postsReduser;
