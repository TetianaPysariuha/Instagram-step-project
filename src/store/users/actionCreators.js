import {
  LOGGING_USER, GET_USERS, GET_USER_BY_ID, ADD_USER, UPDATE_USER, DELETE_USER, GET_SUBSCRIBERS_BY_USER_ID,
} from './actiions';

const SERVER_URL = 'https://nameless-lake-66137.herokuapp.com';

export const loggedUserReceiveData = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: LOGGING_USER, payload: user });
};

export const getUsers = () => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_USERS, payload: users });
};

export const getUserById = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_USER_BY_ID, payload: user });
};

export const addNewUser = (payload) => async (dispatch) => {
  const user = await fetch(`${SERVER_URL}/users`, {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: ADD_USER, payload: user });
};

export const updateUser = (payload) => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users/${payload.id}`, {
    method: 'PUT',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload.data),
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: UPDATE_USER, payload: users });
};

export const deleteUser = (payload) => async (dispatch) => {
  const users = await fetch(`${SERVER_URL}/users/${payload}`, {
    method: 'DELETE',
    mode: 'cors',
    headers: { 'Content-Type': 'aplication/json' },
  }).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: DELETE_USER, payload: users });
};

export const getSubscribersByUserId = (payload) => async (dispatch) => {
  const subscribers = await fetch(`${SERVER_URL}/users/${payload}`).then((res) => res.json()).then((data) => data.data);
  dispatch({ type: GET_SUBSCRIBERS_BY_USER_ID, payload: subscribers });
};
