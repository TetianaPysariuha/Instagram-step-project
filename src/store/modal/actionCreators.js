import { OPEN_MODAL, CLOSE_MODAL } from './actiions';

export const openModalAC = (payload) => ({ type: OPEN_MODAL, payload });
export const closeModalAC = () => ({ type: CLOSE_MODAL });
