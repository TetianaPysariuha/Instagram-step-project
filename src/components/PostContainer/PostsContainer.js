/* eslint-disable no-underscore-dangle */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import styles from './PostsContainer.module.scss';
import Post from '../Post/Post';
import UserName from '../UserName/UserName';
import { updatePost, showMoreChange } from '../../store/posts/actionCreators';

function PostsContainer(props) {
  const dispatch = useDispatch();
  const { posts } = props;
  const postsStore = useSelector((store) => store.posts.posts);
  const showMoreComments = useSelector((store) => store.posts.showMoreComments);
  const loggedUser = useSelector((store) => store.users.loggedUser);
  const users = useSelector((store) => store.users.users);

  const postsList = posts || filteredPosts;
  const filteredPosts = postsList.filter((post) => post.userid !== loggedUser._id);

  const handleCklickLike = (postId, userId) => {
    const post = postsList.find((el) => el._id === postId);
    let newLikes;
    if (post.likes.includes(userId)) {
      newLikes = post.likes.filter((el) => el !== userId);
      dispatch(updatePost({ id: postId, data: { likes: newLikes } }));
    } else {
      newLikes = [...post.likes];
      newLikes.push(userId);
    }
    dispatch(updatePost({ id: postId, data: { likes: newLikes } }));
  };

  const handleCklickFavorite = (postId, userId) => {
    const post = postsList.find((el) => el._id === postId);
    let newFavorite;
    if (post.favorite.includes(userId)) {
      newFavorite = post.favorite.filter((el) => el !== userId);
    } else {
      newFavorite = [...post.favorite];
      newFavorite.push(userId);
    }
    dispatch(updatePost({ id: postId, data: { favorite: newFavorite } }));
  };

  const handleCklickShowMore = (postId) => {
    dispatch(showMoreChange({ postId }));
  };

  const createCommentElement = (postId, comments) => {
    if (showMoreComments.includes(postId)) {
      return (
        <ul>
          {comments.map((el) => (
            <li key={el._id}>
              <span>{users.find((elem) => elem._id === el.userId).nik}</span>
              {' '}
              {el.text}
            </li>
          ))}
        </ul>
      );
    }
    return (
      <ul>
        <li key={comments._id}>
          <span>{users.find((elem) => elem._id === comments[comments.length - 1].userId).nik}</span>
          {' '}
          {comments[comments.length - 1].text}
        </li>
      </ul>
    );
  };

  const getPostElement = (post) => {
    const {
      _id: postId, user: { avatar, nik }, img, title, description,
    } = post;
    const isLiked = post.likes.includes(loggedUser._id);
    const isFavorite = post.favorite.includes(loggedUser._id);
    let comments = null;
    if (post.comments.length > 0) {
      comments = createCommentElement(postId, post.comments);
    }
    const isMore = !(post.comments.length > 1 && showMoreComments.includes(postId));

    return (
      <Post
        key={postId}
        postId={postId}
        userName={<UserName image={avatar} nickname={nik} additionalString={title} />}
        mainImg={img}
        title={title}
        description={description}
        isLiked={isLiked}
        handleCklickLike={() => handleCklickLike(postId, loggedUser._id)}
        handleCklickComments={() => {}}
        isFavorite={isFavorite}
        handleCklickFavorite={() => handleCklickFavorite(postId, loggedUser._id)}
        comments={comments}
        isMore={isMore}
        handleCklickShowMore={() => handleCklickShowMore(postId)}
      />
    );
  };

  return (
    <div className={styles.postContainer}>
      {postsList.length > 0 && users.length > 0 && postsList.map((post) => getPostElement(post))}
    </div>
  );
}

PostsContainer.prototype = {
  posts: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string,
    img: PropTypes.string,
    userid: PropTypes.string,
    description: PropTypes.string,
    likes: PropTypes.arrayOf(PropTypes.string),
    favorite: PropTypes.arrayOf(PropTypes.string),
    comments: PropTypes.arrayOf(PropTypes.shape({
      _id: PropTypes.string,
      userid: PropTypes.string,
      text: PropTypes.string,
    })),
    seeCount: PropTypes.number,
    __v: PropTypes.number,
    user: PropTypes.arrayOf(PropTypes.shape({
      _id: PropTypes.string,
      avatar: PropTypes.string,
      nik: PropTypes.string,
      name: PropTypes.string,
      followBy: PropTypes.arrayOf(PropTypes.string),
      __v: PropTypes.number,
    })),
  })),
};

Post.defaultProps = {
  posts: [],
};

export default PostsContainer;
