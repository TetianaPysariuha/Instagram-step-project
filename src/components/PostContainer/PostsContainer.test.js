/* eslint-disable no-unused-vars */
import { render, screen, fireEvent } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import PostsContainer from './PostsContainer';
import * as actions from '../../store/posts/actionCreators';
import '@testing-library/jest-dom';

jest.mock('react-redux');

const posts = [{
  _id: '63558b3d161b35a6d8b4d9ff',
  img: 'https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg',
  userid: '635429e5461d59d5a106db66',
  title: 'Some post for deleting user',
  description: 'Some description about post',
  likes: [],
  favorite: [],
  comments: [],
  seeCount: 999,
  __v: 0,
  user: {
    _id: '635429e5461d59d5a106db66',
    avatar: 'https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755687/instagram/avatar/1_wfz4lc.jpg',
    nik: 'Milda',
    name: 'Patrisia Mumamba',
    followBy: [
      '6355802bab5157181c8ed0d2',
      '6355802bab5157181c8ed0d4',
      '6355802bab5157181c8ed0d5',
      '6355802bab5157181c8ed0d3',
      '635ad2dece62ea7848f38ea9',
      '6355802bab5157181c8ed0da',
    ],
    __v: 0,
  },
}];
const user = {
  _id: '635429e5461d59d5a106db66',
  avatar: 'https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755687/instagram/avatar/1_wfz4lc.jpg',
  nik: 'Milda',
  name: 'Patrisia Mumamba',
  followBy: [
    '6355802bab5157181c8ed0d2',
    '6355802bab5157181c8ed0d4',
    '6355802bab5157181c8ed0d5',
    '6355802bab5157181c8ed0d3',
    '635ad2dece62ea7848f38ea9',
    '6355802bab5157181c8ed0da',
  ],
};
const usersArray = [user];

describe('Post snapshot testing', () => {
/*   test('Should PostContainer to render without props', () => {
    const postsStore = useSelector.mockReturnValue(posts);
    const showMoreComments = useSelector.mockReturnValue([]);
    const loggedUser = useSelector.mockReturnValue(user);
    const users = useSelector.mockReturnValue(usersArray);

    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    const { asFragment } = render(
      <BrowserRouter>
        <PostsContainer />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  }); */

  test('Should PostContainer to render with props', () => {
    const postsStore = useSelector.mockReturnValue(posts);
    const showMoreComments = useSelector.mockReturnValue([]);
    const loggedUser = useSelector.mockReturnValue(user);
    const users = useSelector.mockReturnValue(usersArray);

    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    const { asFragment } = render(
      <BrowserRouter>
        <PostsContainer posts={posts} />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('PostContainer functions work', () => {
  test('Click on button Like', () => {
    const handleCklickLike = jest.spyOn(actions, 'updatePost');
    const showMoreComments = useSelector.mockReturnValue([]);
    const loggedUser = useSelector.mockReturnValue(user);
    const users = useSelector.mockReturnValue(usersArray);

    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    render(
      <BrowserRouter>
        <PostsContainer posts={posts} />
      </BrowserRouter>,
    );

    fireEvent.click(screen.getByTestId('likeBtn'));
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(handleCklickLike).toHaveBeenCalledTimes(1);
  });

  test('Click on button favorite', () => {
    const handleCklickFavorite = jest.spyOn(actions, 'updatePost');
    const showMoreComments = useSelector.mockReturnValue([]);
    const loggedUser = useSelector.mockReturnValue(user);
    const users = useSelector.mockReturnValue(usersArray);

    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    render(
      <BrowserRouter>
        <PostsContainer posts={posts} />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByTestId('favoriteBtn'));
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(handleCklickFavorite).toHaveBeenCalledTimes(1);
  });

  test('Click on button ShowMore', () => {
    const handleCklickShowMore = jest.spyOn(actions, 'showMoreChange');
    const showMoreComments = useSelector.mockReturnValue([]);
    const loggedUser = useSelector.mockReturnValue(user);
    const users = useSelector.mockReturnValue(usersArray);

    const dispatch = jest.fn();
    useDispatch.mockReturnValue(dispatch);

    render(
      <BrowserRouter>
        <PostsContainer posts={posts} />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Show more comments', { exact: false }));
    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(handleCklickShowMore).toHaveBeenCalledTimes(1);
  });
});
