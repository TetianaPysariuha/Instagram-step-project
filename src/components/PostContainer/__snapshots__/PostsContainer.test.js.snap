// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post snapshot testing Should PostContainer to render with open Modal 1`] = `
<DocumentFragment>
  <div
    class="postContainer"
  >
    <div
      class="postContainier"
    >
      <div
        class="postWrapper"
      >
        <div
          class="userNameContainer"
        >
          <a
            class="avatarLink"
            href="/userpage/635429e5461d59d5a106db66"
          >
            <img
              alt="avatar"
              class="avatar"
              src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755687/instagram/avatar/1_wfz4lc.jpg"
            />
          </a>
          <div
            class="nicknameWrraper"
          >
            <div>
              <a
                class="nickname"
                href="/userpage/635429e5461d59d5a106db66"
              >
                <span>
                  Milda
                </span>
              </a>
              <p
                class="comment"
              />
            </div>
            <p
              class="additionalString"
            >
              Some post for deleting user
            </p>
          </div>
        </div>
        <div
          class="maneFoto"
          style="background-image: url(https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg);"
        >
          <img
            alt="Some post for deleting user"
            src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg"
          />
        </div>
        <div
          class="buttonBox"
        >
          <div
            class="groupBtn"
          >
            <img
              alt="not liked"
              data-testid="likeBtn"
              src="../../images/heart-white.svg"
            />
          </div>
          <img
            alt="favorite"
            data-testid="favoriteBtn"
            src="../../images/favorite.svg"
          />
        </div>
        <p
          class="description"
        >
          Some description about post
        </p>
        <div
          class="comments"
        >
          <ul>
            <li>
              <span>
                Milda
              </span>
               zxcvxzcv
            </li>
            <li>
              <span>
                Milda
              </span>
               asdasdasd
            </li>
            <li>
              <span>
                Milda
              </span>
               xzzzxcxzc
            </li>
          </ul>
        </div>
        <div
          class="addComment"
        >
          <form
            action="#"
            class="form"
          >
            <input
              class="formField"
              name="comment"
              placeholder="Напишіть коментар"
              type="text"
              value=""
            />
            <button
              class="btnSubmit"
              disabled=""
              type="submit"
            >
              Опублікувати
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Post snapshot testing Should PostContainer to render with props 1`] = `
<DocumentFragment>
  <div
    class="postContainer"
  >
    <div
      class="postContainier"
    >
      <div
        class="postWrapper"
      >
        <div
          class="userNameContainer"
        >
          <a
            class="avatarLink"
            href="/userpage/635429e5461d59d5a106db66"
          >
            <img
              alt="avatar"
              class="avatar"
              src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755687/instagram/avatar/1_wfz4lc.jpg"
            />
          </a>
          <div
            class="nicknameWrraper"
          >
            <div>
              <a
                class="nickname"
                href="/userpage/635429e5461d59d5a106db66"
              >
                <span>
                  Milda
                </span>
              </a>
              <p
                class="comment"
              />
            </div>
            <p
              class="additionalString"
            >
              Some post for deleting user
            </p>
          </div>
        </div>
        <div
          class="maneFoto"
          style="background-image: url(https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg);"
        >
          <img
            alt="Some post for deleting user"
            src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg"
          />
        </div>
        <div
          class="buttonBox"
        >
          <div
            class="groupBtn"
          >
            <img
              alt="not liked"
              data-testid="likeBtn"
              src="../../images/heart-white.svg"
            />
            <img
              alt="see all comments"
              data-testid="commentBtn"
              src="../../images/comment.svg"
            />
          </div>
          <img
            alt="favorite"
            data-testid="favoriteBtn"
            src="../../images/favorite.svg"
          />
        </div>
        <p
          class="description"
        >
          Some description about post
        </p>
        <div
          class="comments"
        >
          <ul>
            <li>
              <span>
                Milda
              </span>
               xzzzxcxzc
            </li>
          </ul>
          <button
            class="showMoreBtn"
            data-testid="showMoreBtn"
            type="button"
          >
            Показати більше
          </button>
        </div>
        <div
          class="addComment"
        >
          <form
            action="#"
            class="form"
          >
            <input
              class="formField"
              name="comment"
              placeholder="Напишіть коментар"
              type="text"
              value=""
            />
            <button
              class="btnSubmit"
              disabled=""
              type="submit"
            >
              Опублікувати
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Post snapshot testing Should PostContainer to render without props 1`] = `
<DocumentFragment>
  <div
    class="postContainer"
  >
    <div
      class="postContainier"
    >
      <div
        class="postWrapper"
      >
        <div
          class="userNameContainer"
        >
          <a
            class="avatarLink"
            href="/userpage/635429e5461d59d5a106db66"
          >
            <img
              alt="avatar"
              class="avatar"
              src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755687/instagram/avatar/1_wfz4lc.jpg"
            />
          </a>
          <div
            class="nicknameWrraper"
          >
            <div>
              <a
                class="nickname"
                href="/userpage/635429e5461d59d5a106db66"
              >
                <span>
                  Milda
                </span>
              </a>
              <p
                class="comment"
              />
            </div>
            <p
              class="additionalString"
            >
              Some post for deleting user
            </p>
          </div>
        </div>
        <div
          class="maneFoto"
          style="background-image: url(https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg);"
        >
          <img
            alt="Some post for deleting user"
            src="https://res.cloudinary.com/dx84fdyhd/image/upload/v1667755090/instagram/post14_dmu9vm.jpg"
          />
        </div>
        <div
          class="buttonBox"
        >
          <div
            class="groupBtn"
          >
            <img
              alt="not liked"
              data-testid="likeBtn"
              src="../../images/heart-white.svg"
            />
            <img
              alt="see all comments"
              data-testid="commentBtn"
              src="../../images/comment.svg"
            />
          </div>
          <img
            alt="favorite"
            data-testid="favoriteBtn"
            src="../../images/favorite.svg"
          />
        </div>
        <p
          class="description"
        >
          Some description about post
        </p>
        <div
          class="comments"
        >
          <ul>
            <li>
              <span>
                Milda
              </span>
               xzzzxcxzc
            </li>
          </ul>
          <button
            class="showMoreBtn"
            data-testid="showMoreBtn"
            type="button"
          >
            Показати більше
          </button>
        </div>
        <div
          class="addComment"
        >
          <form
            action="#"
            class="form"
          >
            <input
              class="formField"
              name="comment"
              placeholder="Напишіть коментар"
              type="text"
              value=""
            />
            <button
              class="btnSubmit"
              disabled=""
              type="submit"
            >
              Опублікувати
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
